itou/employee_record/tests/conftest.py

```diff
@@ -0,0 +1,138 @@
+        except OSError as e:
+            return paramiko.SFTPServer.convert_errno(e.errno)
+
+        handle = paramiko.SFTPHandle(flags)
+        handle.filename = path
+        handle.readfile = f
+        handle.writefile = f
+        return handle
+
+    def remove(self, path):
+        try:
+            os.remove(self._realpath(path))
+        except OSError as e:
+            return paramiko.SFTPServer.convert_errno(e.errno)
+        return paramiko.SFTP_OK
+
+
+@pytest.fixture(name="sftp_directory")
+def sftp_directory_fixture(tmp_path_factory):
+    return tmp_path_factory.mktemp("sftp")
+
+
+@pytest.fixture(name="sftp_client_factory")
+def sftp_client_factory_fixture(sftp_directory):
+    """
+    Set up an in-memory SFTP server thread. Return the client Transport/socket.
+
+    The resulting client Transport (along with all the server components) will
+    be the same object throughout the test session; the `sftp_client_factory` fixture then
+    creates new higher level client objects wrapped around the client Transport, as necessary.
+    """
+    # Sockets & transports
+    server_socket, client_socket = socket.socketpair()
+    server_transport = paramiko.Transport(server_socket)
+    client_transport = paramiko.Transport(client_socket)
+
+    # Auth
+    host_key = """
+-----BEGIN OPENSSH PRIVATE KEY-----
+b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAlwAAAAdzc2gtcn
+NhAAAAAwEAAQAAAIEApv6JQanhstho5tnOTxqn4WHt9mxNxp+jwKsYN97par77LZ0mUcbq
+YmJk4Hys3X3lO9WGMzmVzRHEl2fDGYW1vEvhCJv1PsgVCJQvSIj8SNCHuAc+EgNRoeEdEZ
+zk13gvdk/PK4S16hyZJIyVKpZYyW69dsPZ6yeXIM2U80ebUK0AAAIIRVmdIEVZnSAAAAAH
+c3NoLXJzYQAAAIEApv6JQanhstho5tnOTxqn4WHt9mxNxp+jwKsYN97par77LZ0mUcbqYm
+Jk4Hys3X3lO9WGMzmVzRHEl2fDGYW1vEvhCJv1PsgVCJQvSIj8SNCHuAc+EgNRoeEdEZzk
+13gvdk/PK4S16hyZJIyVKpZYyW69dsPZ6yeXIM2U80ebUK0AAAADAQABAAAAgAcIYS9qwB
+6QDprqQU4DiV5siWD/XqcmwfazPenccWgVcu6Q17RJ9EUAb88kcQNC8gx7CZPOsQhZ3zWh
+IQ5KG0UpYS8Dpllf/WfqFmJ7d8At9+2SvGO/5fhVwzL/3oFT86w5bIjL6W6jQtMScAVHKz
+MdjKjxdSQBf2IH0bbTIxcBAAAAQExhCewI5iDXSgSB9R5cdtK51EGVFGnBQLMEEjJ+z4QL
+P4+utD1165J6cJLlsUkaCjVCB3Dv4vEIAJpwnveqWn8AAABBANQLq3Db5Bx/zLXR4ResNN
+i6zUhVuu6O4ct8+79ch2bWl4An1M289xFKrwUznxVEpy57Nih8rn4ra3scqXndAtkAAABB
+AMmcMPC5KpepRwvJUQKLGh2iSvZgsHgIerrFHb46SedUZt8qeM6CuvTo9Kt0Hz5AcZQhAD
+mYF6/I4G8PybmH7/UAAAATa2V5IGZvciB0ZXN0IHNlcnZlcg==
+-----END OPENSSH PRIVATE KEY-----
+    """
+    server_transport.add_server_key(paramiko.RSAKey.from_private_key(io.StringIO(host_key)))
+
+    # Server setup
+    server_transport.set_subsystem_handler("sftp", paramiko.SFTPServer, RootedSFTPServer, root_path=sftp_directory)
+    # The event parameter is here to not block waiting for a client connection
+    server_transport.start_server(event=threading.Event(), server=Server())
+
+    client_transport.connect(username="user", password="password")
+
+    def sftp_client_factory(*args, **kwargs):
+        return paramiko.SFTP.from_transport(client_transport)
```

Comment:
```suggestion
        return paramiko.SFTPClient.from_transport(client_transport)
```
cf https://docs.paramiko.org/en/latest/api/sftp.html#paramiko.sftp_client.SFTP

Reply:
Ce qui me perturbe c'est qu'on va pouvoir [créer](https://docs.paramiko.org/en/latest/api/sftp.html#paramiko.sftp_client.SFTPClient.from_transport) plusieurs `SFTPClient object, referring to an sftp session (channel) across the transport` et que la plupart des tests finiront par appeler [close()](https://docs.paramiko.org/en/latest/api/sftp.html#paramiko.sftp_client.SFTPClient.close) dessus ce qui est censé `Close the SFTP session and its underlying channel` et je m'attendrais donc à ce que la `socketpair` soit également fermée :thinking: